src/main/java/com/karankumar/bookproject/ui/shelf/CustomShelfForm.java

```diff
@@ -154,6 +201,17 @@ private void addClassNameToForm() {
                             ".classList.add('shelfFormOverlay');\n");
     }
 
+    public void editShelf(String chosenShelf) {
+        delete.setVisible(true);
+        deletingShelf = chosenShelf;
+        shelfNameField.setValue(deletingShelf);
+        dialog.open();
+    }
```

Comment:
May be we can add a test for this as well?

Reply:
What should be tested here, just that the shelfNameField and deletingShelf components have their values assigned on opening of the dialog?

Reply:
yes thats correct.